### PR TITLE
Raise IssuedForDelivery event

### PR DESCRIPTION
**Pull request by fleetwoods**
Refactor in prep for SMS

### CODE REVIEW

File: src/main/scala/com/ovoenergy/delivery/service/email/process/EmailDeliveryProcess.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+object EmailDeliveryProcess extends LoggingWithMDC {
+
+  def apply(checkBlackWhiteList: (String) => BlackWhiteList.Verdict,
+            isExpired: Option[String] => Boolean,
+            sendEmail: (ComposedEmail) => Either[DeliveryError, GatewayComm])(
+      composedEmail: ComposedEmail): Either[DeliveryError, GatewayComm] = {
+
+    val traceToken = composedEmail.metadata.traceToken
+
+    def blackWhiteListCheck: Either[DeliveryError, _] = checkBlackWhiteList(composedEmail.recipient) match {
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/delivery/service/email/process/EmailDeliveryProcess.scala]: I'd rather see `Unit` in the result type than `_`. Same for `expiryCheck` below.

File: src/main/scala/com/ovoenergy/delivery/service/kafka/DeliveryServiceGraph.scala [hunk 2]
```diff
@@ -31,17 +39,48 @@ object DeliveryServiceGraph extends LoggingWithMDC {
...
+                failure(composedEvent, deliveryError)
+            }
           case None =>
             log.error(s"Skipping event: $msg, failed to parse")
             Future.successful(())
         }
-        result.flatMap(_ => msg.committableOffset.commitScaladsl())
+        result
+          .flatMap(_ => msg.committableOffset.commitScaladsl())
+          .recover { case NonFatal(_) => msg.committableOffset.commitScaladsl() }
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/delivery/service/kafka/DeliveryServiceGraph.scala]: Add some logging inside this recover?

File: src/main/scala/com/ovoenergy/delivery/service/kafka/process/FailedEvent.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    val event = Failed(
+      metadata = Metadata.fromSourceMetadata("delivery-service", comm.metadata),
+      internalMetadata = comm.internalMetadata,
+      reason = deliveryError.description,
+      errorCode = deliveryError.errorCode
+    )
+
+    publishEvent(event).map(record => {
+      logInfo(event.metadata.traceToken,
+              s"Publishing Failed event: ${event.errorCode} - ${event.reason} - ${record.partition}/${record.offset}")
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/delivery/service/kafka/process/FailedEvent.scala]: This log is really useful, we should do this whenever we publish a Failed event across all of our services

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 143](https://circleci.com/gh/ovotech/comms-delivery-service/143) at 2017-03-23 13:28 GMT.

**Pull request merged**